ext/spl/spl_iterators.c

```diff
@@ -485,11 +485,12 @@ static void spl_recursive_it_it_construct(INTERNAL_FUNCTION_PARAMETERS, zend_cla
 
 	switch (rit_type) {
 		case RIT_RecursiveTreeIterator: {
-			zval caching_it_flags, *user_caching_it_flags = NULL;
+			zval caching_it_flags;
+			zend_long user_caching_it_flags = 0;
```

Comment:
You should set it to `= CIT_CATCH_GET_CHILD` here and drop the if below. Otherwise explicitly passing a zero won't be handled correctly.